pkg/processor/runtime/golang/runtime.go

```diff
@@ -100,7 +104,35 @@ func (g *golang) ProcessBatch(batch []nuclio.Event, functionLogger logger.Logger
 	return nil, nuclio.ErrNotImplemented
 }
 
-func (g *golang) callEntrypoint(event nuclio.Event, functionLogger logger.Logger) (response interface{}, responseErr error) {
+func (g *golang) Restart() error {
+	if err := g.Stop(); err != nil {
+		return errors.Wrap(err, "Failed to stop golang runtime")
+	}
+	g.SetStatus(status.Ready)
+	return nil
+}
+
+func (g *golang) Stop() error {
+	g.SetStatus(status.Stopped)
+
+	select {
+	case cancelEventHandling := <-g.cancelEventHandlingChan:
+		cancelEventHandling()
+	default:
+	}
+
+	return nil
+}
+
+// SupportsRestart returns true if the runtime supports restart
+func (g *golang) SupportsRestart() bool {
+	return true
+}
+
+func (g *golang) callEntrypoint(event nuclio.Event, functionLogger logger.Logger) (interface{}, error) {
+	if currentStatus := g.GetStatus(); currentStatus != status.Ready {
+		return nil, errors.Errorf("Runtime not ready (current status: %s)", currentStatus)
+	}
```

Comment:
Adding condition to a critical flow might cause performance hits.
Can you please run benchmark tests to compare the regular flow of with and without the changes?

Reply:
Sure, I will do benchmarking. 

But anyway, 
1. we already have this condition in other runtimes
2. we will need this condition when we start implementing explicitAck/Termination for go runtime

Reply:
@TomerShor results:

1000 total/100 parall:
before: 888.37955ms
after: 841.544612ms

10000 total/100 parall:
before: 955.145597ms
after: 894.964659ms

1000 total/1000 parall:
before:  4.743075748s
after: 4.772192623s

10000 total/1000 parall:
before:  8.847363427s
after: 8.617966213s

Reply:
probably, moving of duration metrics calculation could impact performance, so we can see insignificant improve in some cases